src/handler.ts

```diff
@@ -100,28 +91,36 @@ abstract class AbstractFileHandler implements IMessageFilter {
    * contents of the widget.
    */
   get finished(): ISignal<AbstractFileHandler, IContentsModel> {
-    return AbstractFileHandler.finishedSignal.bind(this);
+    return Private.finishedSignal.bind(this);
+  }
+
+  /**
+   * A signal emitted when the file handler is activated.
+   */
+  get activated(): ISignal<AbstractFileHandler, void> {
+    return Private.activatedSignal.bind(this);
   }
 
   /**
-   * Open a path and return a widget.
+   * Deactivate the handler.
+   */
+  deactivate(): void {
+    this._activeWidget = null;
+  }
+
+  /**
+   * Open a contents model and return a widget.
    */
   open(model: IContentsModel): Widget {
-    let path = model.path;
-    let index = arrays.findIndex(this.widgets,
-      (widget, ind) => {
-        return AbstractFileHandler.modelProperty.get(widget).path === path;
-      }
-    );
-    if (index !== -1) {
-      return this.widgets[index];
+    let widget = this._findWidgetByModel(model);
+    if (!widget) {
+      widget = this.createWidget(model);
+      widget.title.closable = true;
+      widget.title.changed.connect(this.titleChanged, this);
```

Comment:
`s/titleChanged/onTitleChanged`  `titleChanged` looks like a signal to me, whereas `onTitleChanged` looks like a handler.